k8s settings: kube-reserved, eviction-hard, cpu-manager-policy, allow-unsafe-sysctls (#1388)
bottlerocket-os/bottlerocket: README.md

* `settings.kubernetes.standalone-mode`: Whether to run the kubelet in standalone mode, without connecting to an API server.  Defaults to `false`.
* `settings.kubernetes.authentication-mode`: Which authentication method the kubelet should use to connect to the API server, and for incoming requests.  Defaults to `aws` for AWS variants, and `tls` for other variants.
* `settings.kubernetes.bootstrap-token`: The token to use for [TLS bootstrapping](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/).  This is only used with the `tls` authentication mode, and is otherwise ignored.

> * `settings.kubernetes.eviction-hard`: The hard eviction Signal and Thresholds you set up to reclaim the associated starved resource.

Reviewer: ```suggestion
* `settings.kubernetes.eviction-hard`: The signals and thresholds that trigger pod eviction.
```

  Remember to quote keys (since they often contain ".") and to quote all values.
  * Example user data for setting up eviction hard:
    ```
    [settings.kubernetes.eviction-hard]
    "memory.available" = "15%"
    "nodefs.inodesFree" = "15Mi"

>     "pid.available" = "15%"

Reviewer: It might be better to have just one example document, potentially in a different file, to avoid cluttering the main README with specifics about settings that aren't widely used.

Author: Sure thing!

* `settings.kubernetes.standalone-mode`: Whether to run the kubelet in standalone mode, without connecting to an API server.  Defaults to `false`.
* `settings.kubernetes.authentication-mode`: Which authentication method the kubelet should use to connect to the API server, and for incoming requests.  Defaults to `aws` for AWS variants, and `tls` for other variants.
* `settings.kubernetes.bootstrap-token`: The token to use for [TLS bootstrapping](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/).  This is only used with the `tls` authentication mode, and is otherwise ignored.
* `settings.kubernetes.eviction-hard`: The signals and thresholds that trigger pod eviction.

>   Remember to quote keys (since they often contain ".") and to quote all values.

Reviewer: It looks like all signals are named in "x.y" form, so:

```suggestion
  Remember to quote signals (since they all contain ".") and to quote all values.
```

* `settings.kubernetes.cluster-dns-ip`: The CIDR block of the primary network interface.
* `settings.kubernetes.node-ip`: The IPv4 address of this node.
* `settings.kubernetes.pod-infra-container-image`: The URI of the "pause" container.
* `settings.kubernetes.kube-reserved`: Resources reserved for node components.

>   * Bottlerocket provides default values for the resources. Default values are taken from [EKS' calculations](https://github.com/awslabs/amazon-eks-ami/blob/db28da15d2b696bc08ac3aacc9675694f4a69933/files/bootstrap.sh):

Reviewer: I would prefer to link to our own implementation within the Bottlerocket repo.

>     `cpu`: in millicores from the total number of vCPUs available on the instance.

Reviewer 2: If you intend for these to be bullets, you need to add a "*" in front of each bullet point. You can check the richdiff by clicking on the page icon in the upper right corner when viewing diffs.

Author: yeah. I will fix in next push
